python/python/bystro/ancestry/listener.py

```diff
@@ -0,0 +1,105 @@
+"""Provide a worker for the ancestry model."""
+import argparse
+import logging
+from pathlib import Path
+
+from ruamel.yaml import YAML
+
+from bystro.ancestry.ancestry_types import AncestryResponse, AncestrySubmission
+from bystro.beanstalkd.messages import BaseMessage, CompletedJobMessage, SubmittedJobMessage
+from bystro.beanstalkd.worker import ProgressPublisher, QueueConf, get_progress_reporter, listen
+
+logger = logging.getLogger()
+
+ANCESTRY_TUBE = "ancestry"
+
+
+class AncestryJobData(BaseMessage, frozen=True):
+    """Wrap an AncestrySubmission in a BaseMessage for beanstalk."""
+
+    ancestry_submission: AncestrySubmission
+
+
+class AncestryJobCompleteMessage(CompletedJobMessage, frozen=True, kw_only=True):
+    """Wrap an AncestryResponse in a CompletedJobMessage for beanstalk."""
+
+    results: AncestryResponse
+
+
+def _load_queue_conf(queue_conf_path: str) -> QueueConf:
+    with Path(queue_conf_path).open(encoding="utf-8") as queue_config_file:
+        raw_queue_conf = YAML(typ="safe").load(queue_config_file)
+    beanstalk_conf = raw_queue_conf["beanstalkd"]
+    return QueueConf(addresses=beanstalk_conf["addresses"], tubes=beanstalk_conf["tubes"])
+
+
+def _infer_ancestry(
+    ancestry_submission: AncestrySubmission, _publisher: ProgressPublisher
+) -> AncestryResponse:
+    """Run an ancestry job."""
+    # TODO: main ancestry model logic goes here.  Just stubbing out for now.
+    logger.debug("Inferring ancestry for: %s", ancestry_submission)
+
+    # not doing anything with this reporter at the moment, we're
+    # simply threading it through for later.
+    _reporter = get_progress_reporter(_publisher)
+    vcf_path = ancestry_submission.vcf_path
+    return AncestryResponse(vcf_path=vcf_path, results=[])
+
+
+async def handler_fn(
+    publisher: ProgressPublisher, ancestry_job_data: AncestryJobData
+) -> AncestryResponse:
+    """Do ancestry job, wraping _infer_ancestry for beanstalk."""
+    # Separating _handler_fn from _infer_ancestry in order to separate ML from infra concerns,
+    # and especially to keep _infer_ancestry eager.
+    return _infer_ancestry(ancestry_job_data.ancestry_submission, publisher)
+
+
+def submit_msg_fn(ancestry_job_data: AncestryJobData) -> SubmittedJobMessage:
+    """Acknowledge receipt of AncestryJobData."""
+    return SubmittedJobMessage(ancestry_job_data.submissionID)
```

Comment:
Here and throughout, are we handling submissionIDs correctly?  Would there be any way to make that harder to mess up?

Reply:
Ok, I just wanted to be sure the worker was just supposed to return the original submissionID, and wasn't responsible for altering / replacing it in any way.

At some point it would be a big help to have some docs on submissionID / queueID semantics for future reference, maybe in the beanstalkd directory.

As an aside, I do think that long-term we should be running with `mypy --strict` tbh, or at least a stricter config containing the rules we think give us most bang for our buck, including `disallow_untyped_def`, but I guess that's out of scope here.